prospero-cli/src/main/java/org/wildfly/prospero/cli/commands/channel/ChannelInitializeCommand.java

```diff
@@ -0,0 +1,175 @@
+        sortOptions = false
+)
+public class ChannelInitializeCommand extends AbstractCommand {
+    public static final String CUSTOMIZATION_REPO_ID = "customization-repository";
+    public static final String CUSTOM_CHANNELS_GROUP_ID = "custom.channels";
+    public static final String DEFAULT_CUSTOMIZATION_REPOSITORY = "customization-repository";
+    @CommandLine.Option(
+            names = CliConstants.CUSTOMIZATION_CHANNEL_NAME
+    )
+    private Optional<String> name;
+
+    @CommandLine.Option(
+            names = CliConstants.CUSTOMIZATION_REPOSITORY_URL,
+            descriptionKey = "customization-repository"
+    )
+    private Optional<URL> repositoryUrl;
+
+    @CommandLine.Option(names = CliConstants.DIR)
+    Optional<Path> directory;
+
+
+    public ChannelInitializeCommand(Console console, ActionFactory actionFactory) {
+        super(console, actionFactory);
+    }
+
+    @Override
+    public Integer call() throws Exception {
+        Path installationDirectory = determineInstallationDirectory(directory);
+        MetadataAction metadataAction = actionFactory.metadataActions(installationDirectory);
+
+        if (!validateRepository(metadataAction) || !validateChannel()) {
+            return ReturnCodes.INVALID_ARGUMENTS;
+        }
+
+        final URL url;
+        if (repositoryUrl.isPresent()) {
+            url = repositoryUrl.get();
+            if (repositoryUrl.get().getProtocol().equals("file")) {
+                if (!createLocalRepository(Paths.get(repositoryUrl.get().toURI()))) {
+                    return ReturnCodes.PROCESSING_ERROR;
+                }
+            }
+        } else {
+            final Path defaultLocalRepoPath = installationDirectory.resolve(InstallationMetadata.METADATA_DIR)
+                    .resolve(DEFAULT_CUSTOMIZATION_REPOSITORY);
+            if (!createLocalRepository(defaultLocalRepoPath)) {
+                return ReturnCodes.PROCESSING_ERROR;
+            }
+            url = defaultLocalRepoPath.toUri().toURL();
+        }
+
+        if (name.isEmpty()) {
+            if (metadataAction.getChannels().stream().anyMatch(c->c.getGav()!=null && c.getGav().startsWith(CUSTOM_CHANNELS_GROUP_ID + ":"))) {
+                console.error(CliMessages.MESSAGES.customizationChannelAlreadyExists());
+                return ReturnCodes.PROCESSING_ERROR;
+            }
+            name = Optional.of(CUSTOM_CHANNELS_GROUP_ID + ":" + RandomStringUtils.randomAlphanumeric(8));
+        }
+
+        // add new channel
+        console.println(CliMessages.MESSAGES.registeringCustomChannel(name.get()));
+        metadataAction.addChannel(name.get());
+
+        // add new repository
+        console.println(CliMessages.MESSAGES.registeringCustomRepository(url.toString()));
+        metadataAction.addRepository(CUSTOMIZATION_REPO_ID, url);
+        console.println(CliMessages.MESSAGES.repositoryAdded(CUSTOMIZATION_REPO_ID));
+
+        // if the url is local folder, make sure it's created
```

Comment:
This comment is "TODO:" or is a forgotten line?

Reply:
old comment, I'll remove it